src/custom_gates.jl

```diff
@@ -18,6 +30,92 @@ function matrix_rep(g::DoubleExcitation)
     return SMatrix{16,16,ComplexF64}(mat)
 end
 
+@doc raw"""
+    DoubleExcitationPlus(ϕ)
+
+Generate the matrix representation of the [DoubleExcitationPlus](https://docs.pennylane.ai/en/stable/code/api/pennylane.DoubleExcitationPlus.html) gate.
+
+This operation performs an SO(2) rotation in the two-dimensional subspace {|1100\rangle, |0011\rangle}
+while applying a phase-shift on other states. More precisely, it performs the transformation:
+
+```math
+|0011\rangle & \rightarrow \cos\left(\frac{\phi}{2}\right)|0011\rangle - \sin\left(\frac{\phi}{2}\right)|1100\rangle \\
+|1100\rangle & \rightarrow \cos\left(\frac{\phi}{2}\right)|1100\rangle + \sin\left(\frac{\phi}{2}\right)|0011\rangle \\
+|x\rangle & \rightarrow e^{\frac{i\phi}{2}}|x\rangle \quad \text{for all other basis states } |x\rangle
+```
+"""
+struct DoubleExcitationPlus <: AngledGate{1}
+    angle::NTuple{1,Union{Float64,FreeParameter}}
+    DoubleExcitationPlus(angle::T) where {T<:NTuple{1,Union{Float64,FreeParameter}}} =
+        new(angle)
+end
+Braket.chars(::Type{DoubleExcitationPlus}) = "G2+(ang)"
+Braket.qubit_count(::Type{DoubleExcitationPlus}) = 4
+Base.inv(g::DoubleExcitationPlus) = DoubleExcitationPlus(-g.angle[1])
+function matrix_rep(g::DoubleExcitationPlus)
+    cosϕ = cos(g.angle[1] / 2.0)
+    sinϕ = sin(g.angle[1] / 2.0)
+    eiϕ2 = exp(im * g.angle[1] / 2.0)
+ 
+    mat = diagm(fill(eiϕ2, 16))
```

Comment:
I don't think this is right in the |0011>, |1100> subspace?

Reply:
I included it to cater to applying e^(iϕ/2)|x⟩, for all other basis states |x⟩.

usual details{{
|0011⟩→cos(ϕ/2)|0011⟩−sin(ϕ/2)
|1100⟩|1100⟩→cos(ϕ/2)|1100⟩+sin(ϕ/2)}}

while applying a phase-shift on other states

|x⟩→e^(iϕ/2)|x⟩,

for all other basis states |x⟩

Reply:
In short, We apply the phase shift to all basis states initially and then modify only the {∣1100⟩,∣0011⟩}{∣1100⟩,∣0011⟩} subspace to reflect the SO(2) rotation!

Reply:
Maybe, something like this would make better sense!

```
function matrix_rep(g::DoubleExcitationPlus)
    cosϕ = cos(g.angle[1] / 2.0)
    sinϕ = sin(g.angle[1] / 2.0)
    eiϕ2 = exp(im * g.angle[1] / 2.0)
    mat = diagm(ones(ComplexF64, 16))
    mat[4, 4]   = cosϕ
    mat[13, 13] = cosϕ
    mat[4, 13]  = -sinϕ
    mat[13, 4]  = sinϕ
    # Apply phase-shift to states outside rotation subspace
    for i in 1:16
        if i != 4 && i != 13
            mat[i, i] *= eiϕ2
        end
    end
    return SMatrix{16,16,ComplexF64}(mat)
end
```